src/VirtoCommerce.CatalogModule.Data/Services/CategoryIteratorFactory.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using VirtoCommerce.CatalogModule.Core.Search;
+using VirtoCommerce.CatalogModule.Data.Repositories;
+using VirtoCommerce.Platform.Core.Settings;
+
+namespace VirtoCommerce.CatalogModule.Data.Services
+{
+    public sealed class CategoryIteratorFactory
```

Comment:
rename to CategoryHierarchyIteratorFactory
or consider removing this class and using Func<> factory registration in ServiceCollection